### PR TITLE
refactor: remove explicit helm and stern installation dependencies

### PR DESCRIPTION
**Pull request by edvald**
We now automatically fetch helm when needed via the new tool helper.

Note: We still install Helm via package dependencies, because it's
really no bother for the user, but we don't need to explicitly list
it in docs.

Before completing this and using for more tools, I wanted to float the idea and see if it makes sense to everyone...

### CODE REVIEW

File: garden-cli/src/plugins/kubernetes/helm.ts [hunk 1]
```diff
@@ -221,11 +222,44 @@ async function build({ ctx, module, logEntry }: BuildModuleParams<HelmModule>):
...
+    },
+    linux: {
+      url: "https://storage.googleapis.com/kubernetes-helm/helm-v2.10.0-linux-amd64.tar.gz",
+      sha256: "0fa2ed4983b1e4a3f90f776d08b88b0c73fd83f305b5b634175cb15e61342ffe",
+      extract: {
+        format: "tar",
+        executablePath: ["linux-amd64", "helm"],
+      },
+    },
+    win32: {
```
Comment (ellenkorbes) [inline garden-cli/src/plugins/kubernetes/helm.ts]: Isn't it a bit confusing to call it `win32` when we specifically only allow things to run on x86?

Comment (edvald) [reply garden-cli/src/plugins/kubernetes/helm.ts]: This is standard Node.js terminology (as output by `os.platform()`). And for whatever reason, `win32` is just what it's commonly called, even if it's actually running `x64`

**Comment by ellenkorbes** [comment]
FYI this seems to work fine on Linux.

File: garden-cli/support/install.ps1 [hunk 2]
```diff
@@ -5,12 +5,11 @@
 # 1. Checks whether Hyper-V is enabled.
 # 2. Checks whether Docker is installed.
 # 3. Checks whether Kubernetes is the default orchestrator for Docker.
-# 4. Installs Stern.
-# 5. Check if Chocolatey is installed, installs it if missing.
-# 6. Installs the listed dependencies using Chocolatey if they're not already
+# 4. Check if Chocolatey is installed, installs it if missing.
+# 5. Installs the listed dependencies using Chocolatey if they're not already
 #    present (currently git, nodejs, rsync, and kubernetes-helm).
```
Comment (eysi09) [inline garden-cli/support/install.ps1]: Shouldn't `kubernetes-helm` be removed from this list and from `chocDeps` on [here](https://github.com/garden-io/garden/blob/7bcb7c074f4b273e9191b230fe51981429c36696/garden-cli/support/install.ps1#L141)

Comment (edvald) [reply garden-cli/support/install.ps1]: I debated that, since users are likely to want it installed. But we could remove it, sure.

**Pull request merged**